ChegeBryan/politico
# #163965831 Admin: Register candidate

#### What does this PR do?

Build endpoint to register a candidate for office

#### Description of Task to be completed?

- [x] Add tests for the API endpoint
- [x] Add SQL queries to write and read data from the database
- [x] Add data validations for data

#### How should this be manually tested?
After cloning the repository to your local machine - check
[README.md](README.md) for instructions. Using
[Postman](https://www.getpostman.com/) open a request to create a party [How to create a party](https://github.com/ChegeBryan/politico/pull/45). create an office [How to create an office](https://github.com/ChegeBryan/politico/pull/50). Follow the instructions on the PR's on how to make those requests.
Once you have an office and party also register a user who will be [how to signup user](https://github.com/ChegeBryan/politico/pull/39).
To register a user as candidate take note of the ids returned on the response.

Open a request, add an admin token to the authorization header,  with on this endpoint `POST /api/v2/office/{office_id}/register`
set content type to JSON and add the following in the json body. 
```JSON
{
	"party": 1,
	"candidate": 2
}
```  
#### What are the relevant pivotal tracker stories?
[#163965831](https://www.pivotaltracker.com/story/show/163965831)

#### Screenshots
##### Admin `POST /api/v2/office/{office_id}/register`
![Screenshot from 2019-03-29 01-22-09](https://user-images.githubusercontent.com/40359451/55196682-30285b80-51c1-11e9-874d-518fea3423a9.png)
